src/csharp/Microsoft.Spark/Sql/UDFRegistration.cs

```diff
@@ -241,17 +444,28 @@ public void RegisterJavaUDAF(string name, string className)
         /// <param name="func">Wrapped UDF function</param>
         private void Register<TResult>(string name, Delegate func)
         {
-            Register<TResult>(name, func, UdfUtils.PythonEvalType.SQL_BATCHED_UDF);
+            Register(name, func, UdfUtils.PythonEvalType.SQL_BATCHED_UDF, UdfUtils.GetReturnType(typeof(TResult)));
+        }
+
+        /// <summary>
+        /// Helper function to register wrapped udf.
+        /// </summary>
+        /// <param name="name">Name of the udf</param>
+        /// <param name="func">Wrapped UDF function</param>
+        /// <param name="returnType">Schema associated with the udf</param>
+        private void Register(string name, Delegate func, StructType returnType)
+        {
+            Register(name, func, UdfUtils.PythonEvalType.SQL_BATCHED_UDF, returnType.Json);
         }
 
         /// <summary>
         /// Helper function to register wrapped udf.
         /// </summary>
-        /// <typeparam name="TResult">Return type of the udf</typeparam>
         /// <param name="name">Name of the udf</param>
         /// <param name="func">Wrapped UDF function</param>
         /// <param name="evalType">The EvalType of the function.</param>
-        internal void Register<TResult>(string name, Delegate func, UdfUtils.PythonEvalType evalType)
```

Comment:
Can we leave this API and just call the new one, so that we don't have to change existing calls?
```C#
internal void Register<TResult>(string name, Delegate func, UdfUtils.PythonEvalType evalType)
{
    return Register(name, func, evalType, UdfUtils.GetReturnType(typeof(TResult));
}
```

Reply:
I left the previous internal method as it is, but changed the body to delegate to a newer method which accepts the returnType param.
I have also reverted the changes in the two extension files which called the earlier API
Can you please check and verify if this is ok?